### DISCUSSION TITLE
Instrumentation with opentelemetry

### DISCUSSION THREAD

**Discussion by naartjie** [psycopg/psycopg#390] [General]
Is there any effort under way to instrument versions 3.x with [OpenTelemetry](https://opentelemetry.io/docs/instrumentation/python/)?

I found [this](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/44da3848538b6a756b5edf104a798f538ab0292c/instrumentation/opentelemetry-instrumentation-psycopg2) for psycopg2 but I couldn't tell if there was something which would support psycopg v3.x.

If there is no such project / effort in progress:
- Would it make sense to put such code inside this project itself? Otel is becoming more of a standard and I was recently talking to some of the good folks at honeycomb.io and they suggested it might be something common use libraries might start doing.
- Or does it make more sense to put instrumentation in an external project like [`open-telemetry/opentelemetry-python-contrib`](https://github.com/open-telemetry/opentelemetry-python-contrib)?

**Comment by dvarrazzo** [comment]
Hello,

No, no effort under way in this project. [This module](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-psycopg2/src/opentelemetry/instrumentation/psycopg2) is psycopg2 only, but porting to 3 seems straightforward.

Given that we are not experts in the matter, I believe that the otel python contrib project is a more suitable place.

(converting thus to a discussion, because I don't think we will take actions now, or here, but I'm happy to discuss about anything that might be needed).

**Reply by naartjie** [reply]
Thank you for responding @dvarrazzo and converting to a discussion, I should have opened it as such to start off with.

I'll post updates here if I manage to get anywhere with converting the [psycopg2](https://github.com/open-telemetry/opentelemetry-python-contrib/tree/main/instrumentation/opentelemetry-instrumentation-psycopg2/src/opentelemetry/instrumentation/psycopg2) instrumentation to version 3.

**Reply by adriangb** [reply]
Hi @naartjie did you ever get to working on this? FWIW I think this is a very valuable feature to provide users.

**Reply by naartjie** [reply]
Hey @adriangb. I never did mange to take it any further, nope

**Reply by willianantunes** [reply]
[Just for your information](https://github.com/open-telemetry/opentelemetry-python-contrib/issues/1751).
